src/coprocessor/dag/expr/builtin_other.rs

```diff
@@ -0,0 +1,74 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::i64;
+
+use super::{EvalContext, Result, ScalarFunc};
+use coprocessor::codec::Datum;
+
+impl ScalarFunc {
+    #[inline]
+    pub fn bit_count(&self, ctx: &mut EvalContext, row: &[Datum]) -> Result<Option<i64>> {
+        let res = self.children[0].eval_int(ctx, row);
+        match res {
+            Ok(r) => if let Some(v) = r {
+                Ok(Some(i64::from(v.count_ones())))
+            } else {
+                Ok(None)
+            },
+            Err(e) => if e.is_overflow() {
+                Ok(Some(64))
```

Comment:
I would like to see test cases in some edge cases. In MySQL:

```
mysql> select bit_count(9223372036854775807);
+--------------------------------+
| bit_count(9223372036854775807) |
+--------------------------------+
|                             63 |
+--------------------------------+
1 row in set (0.00 sec)

mysql> select bit_count(9223372036854775808);
+--------------------------------+
| bit_count(9223372036854775808) |
+--------------------------------+
|                              1 |
+--------------------------------+
1 row in set (0.00 sec)

mysql> select bit_count(9223372036854775809);
+--------------------------------+
| bit_count(9223372036854775809) |
+--------------------------------+
|                              2 |
+--------------------------------+
1 row in set (0.00 sec)

mysql> select bit_count(111111111111111111111111111111111111111111111111111111111111111);
+----------------------------------------------------------------------------+
| bit_count(111111111111111111111111111111111111111111111111111111111111111) |
+----------------------------------------------------------------------------+
|                                                                         63 |
+----------------------------------------------------------------------------+
1 row in set, 1 warning (0.01 sec)

mysql> select bit_count('111111111111111111111111111111111111111111111111111111111111111');
+------------------------------------------------------------------------------+
| bit_count('111111111111111111111111111111111111111111111111111111111111111') |
+------------------------------------------------------------------------------+
|                                                                           64 |
+------------------------------------------------------------------------------+
1 row in set, 1 warning (0.01 sec)

mysql> select bit_count(11111111112222222222);
+---------------------------------+
| bit_count(11111111112222222222) |
+---------------------------------+
|                              37 |
+---------------------------------+
1 row in set (0.03 sec)

mysql> select bit_count('11111111112222222222');
+-----------------------------------+
| bit_count('11111111112222222222') |
+-----------------------------------+
|                                37 |
+-----------------------------------+
1 row in set (0.00 sec)
```

Reply:
@breeswish  the above test cases added. PTAL :)